documentation/source/repurchase-agreement-representation.rst

```diff
@@ -0,0 +1,857 @@
+
+
+Introduction
+------------
+The purpose of the CDM for repo and bonds is to provide standardized digital 
+representation of repo products and lifecycle events that will reduce reconciliation 
+failures caused by data and processing mismatches and provides firms with a software 
+toolkit that can be incorporated into existing systems or established as the baseline 
+for a new platform.
+
+Repurchase Agreements are represented in the CDM as contractualProducts
+with an InterestRatePayout and an AssetPayout. Purchase date and repurchase
+date are defined in InterestRatePayout. Margin and haircuts are 
+defined in the collateralProvisions. All other trade related attributes 
+are defined in Trade type.
+
+The types of repo products can be defined is only limited by the structure
+InterestRatePayout and AssetPayout but a a minimum supports fixed term repo,
+open repo, fixed rate and floating rate. 
+
+Repo lifecycle events are supported through a set of functions that accept
+a small set of inputs to auto-generate primitive instructions needed to execute
+business events. Repo lifecycle events include, rolling, re-rating, interium payments, 
+pair-off and shaping.
+
+
+Benefits:
+------------
+
+The benefits of using the CDM for repo transactions is that it provides a 
+standard digital representation of the data required to define a repo product and 
+a standardized set of lifecyle events.
+
+Examples of where the CDM can be used in the repo market:
+
+-  Post-trade matching using a single digital object.
+
+-  Lifecycle event processing across counterparties and settlement services.
+
+-  Internal and external reporting. 
+
+-  Capturing event history.
+
+-  Market standard taxonomy and mapping interface to other platforms.
+
+
+ICMA Contacts:
+------------
+Gabriel Callsen, Director	 
+gabriel.callsen@icmagroup.org
++44 (0)20 7213 0334
+
+Tom Healey, Consultant
+tom.healey@icmagroup.org
+
+
+
+Scope:
+------------
+
+The scope of the CDM Repo initiative was intended to define:
+
+-  Consistent definition of buyer-seller entities based on LEI data
+
+-  Collateral, margin anf haircut attributes.
+
+-  Flexible interest rate payout model to support complex structures.
+
+-  Standardized product taxonomy.
+
+-  Predefine lifecycle event processing and event history.
+
+
+Repo Product Model
+------------------------------
+
+Building upon the ISDA CDM, the Repo CDM added new data types, attributes and events 
+needed to create fixed term, open and floating rate repos, and execute various lifecycle 
+events.
+
+A repo product is formed from a ContractualProduct as an extension of ProductBase. 
+type Product:
+    [metadata key]
+    contractualProduct ContractualProduct (0..1)
+
+At the core of a ContractualProduct is the product’s EconomicTerms that define the 
+attributes that are needed to derive the product’s economic value and determine its cashflows.
+
+.. code-block:: Haskell
+
+type EconomicTerms:
```

Comment:
You need a 1-space indentation for the lines after a code block for the code block to apply. Otherwise it doesn't actually render. So here the first line needs to be " type EconomicTerms:". You can check the rendering directly when editing in GitHub.

Reply:
@tomhealey-icma I have fixed all your indentation issues.